c10/util/AbortHandler.h

```diff
@@ -0,0 +1,80 @@
+#include <c10/util/Backtrace.h>
+#include <c10/util/env.h>
+#include <cstdlib>
+#include <exception>
+#include <iostream>
+#include <mutex>
+
+namespace c10 {
+class AbortHandlerHelper {
+ public:
+  static AbortHandlerHelper& getInstance() {
+#ifdef _WIN32
+    thread_local
+#endif // _WIN32
+        static AbortHandlerHelper instance;
+    return instance;
+  }
+
+  void set(std::terminate_handler handler) {
+    std::lock_guard<std::mutex> lk(mutex);
+    if (!inited) {
```

Comment:
Just curious, shouldn't one raise an exception or something if `set` is called twice?

Reply:
I don't think so, since if someone calls it twice it will do nothing and it is called multiple times anyway since on Windows, it must be called per thread.